ci/bamboo/build-windows.ps1

```diff
@@ -0,0 +1,227 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have purchased from
+# Numenta, Inc. a separate commercial license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+# Builds and tests the nupic.bindings python extension using Release build
+# type.
+#
+# ASUMPTIONS:
+#          1. Building via Numenta Vagrant Task using Vagrant image
+#             bamboo-vagrant-windows-1.x.x.
+#
+#          2. nupic.core root is the current directory on entry.
+#
+#          3. Expects a pristine nupic.core source tree without any remnant
+#             build artifacts from prior build attempts. Otherwise, the
+#             behavior is undefined.
+#
+# NOTES: Based on examples in comments of
+#        https://github.com/NumentaCorp/infrastructure/pull/376
+#
+# OUTPUTS:
+#   nupic.bindings wheel: On success, the resulting wheel will be located in
+#                         the subdirectory nupic_bindings_wheelhouse of the
+#                         source tree's root directory.
+#
+#   Test results: nupic.bindings test results will be located in the
+#                 subdirectory test_results of the source tree's root directory
+#                 with the following content:
+#
+#                 junit-test-results.xml
+#                 htmlcov/
+
+
+# Stop and fail script if any command fails
+$ErrorActionPreference = "Stop"
+
+
+$NupicCoreRootDir = $(get-location).Path
+
+
+
+# Use this function to wrap external commands for powershell error-checking.
+#
+# This is necessary so that `$ErrorActionPreference = "Stop"` will have the
+# desired effect.
+#
+# Returns True if command's $LastExitCode was 0, False otherwise
+#
+# Usage: WrapCmd { cmd arg1 arg2 ... }
+#
+function WrapCmd
+{
+  [CmdletBinding()]
+
+  param (
+    [Parameter(Position=0, Mandatory=1)]
+    [scriptblock]$Command,
+    [Parameter(Position=1, Mandatory=0)]
+    [string]$ErrorMessage = "ERROR: Command failed.`n$Command"
+  )
+  & $Command
+  if ($LastExitCode -eq 0) {
+    return $true
+  }
+  else {
+    Write-Error "WrapCmd: $ErrorMessage"
+    return $false
+  }
+}
+
+
+#
+# Make unix-compatible patch.exe available to the build by copying it from
```

Comment:
@scottpurdy, concerning
> Why do we need patch.exe? Is there something we should do in the setup of the base image to make this available?

I meant to add a comment with a question for you about adding a posix-compliant `patch.exe` to the base image.

I borrowed this code from the AppVeyor script. We need a posix-compliant `patch.exe` because we patch several externals from the cmake-based build. My preference would be to add this command to the base image.

I propose to open a separate ticket for adding posix patch.exe to the base image and to PATH, and then remove this bit of code from build-windows.psi.